### PR TITLE
164798898 configure Eslint

### PR DESCRIPTION
**Pull request by samorahno**
#### Description
A unique style guide needs to be followed, as this would also help the editing process . Integrating Eslint ensures a single coding style is followed.
#### Type of change
- [x]  configuring Eslint
#### Checklist:
- [x]  configure eslint.
- [x]  the general eslint style guide has been included.
#### PIVOTAL BOARD
[#164798898](https://www.pivotaltracker.com/story/show/164798898)

### CODE REVIEW

**Comment by Chiazokam** [comment]
Great work

**Review by davidokonji — APPROVED** [review]
Good work setting ESlint

File: routes/index.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
-const router = require("express").Router();
+const router = require('express').Router();
 
-router.use("/api", require("./api"));
+router.use('/api', require('./api'));
 
 module.exports = router;
```
Comment (paradisekelechi) [inline routes/index.js]: Setting up eslint is not supposed to affect this file.

File: routes/api/users.js [hunk 2]
```diff
@@ -1,87 +1,83 @@
...
-        .then(function() {
-            return res.json({ user: user.toAuthJSON() });
-        })
-        .catch(next);
+  user.save()
+    .then(() => res.json({ user: user.toAuthJSON() }))
+    .catch(next);
 });
 
 module.exports = router;
```
Comment (paradisekelechi) [inline routes/api/users.js]: Setting up eslint is not supposed to affect this file.

File: routes/api/index.js [hunk 3]
```diff
@@ -1,18 +1,18 @@
-const router = require("express").Router();
```
Comment (paradisekelechi) [inline routes/api/index.js]: Setting up eslint is not supposed to affect this file.

File: package.json [hunk 4]
```diff
@@ -28,5 +28,10 @@
     "request": "^2.87.0",
     "slug": "^0.9.1",
     "underscore": "^1.9.1"
+  },
+  "devDependencies": {
+    "eslint": "^5.15.3",
+    "eslint-config-airbnb-base": "^13.1.0",
```
Comment (paradisekelechi) [inline package.json:34]: We are using `airbnb` and not `airbnb-base`

File: models/User.js [hunk 5]
```diff
@@ -1,57 +1,61 @@
-const mongoose = require("mongoose");
```
Comment (paradisekelechi) [inline models/User.js]: Setting up eslint is not supposed to affect this file.

File: index.js [hunk 6]
```diff
@@ -1,93 +1,93 @@
-const fs = require("fs"),
```
Comment (paradisekelechi) [inline index.js]: Setting up eslint is not supposed to affect this file.

File: .eslintrc [hunk 7]
```diff
@@ -0,0 +1,33 @@
+{
+    "root": true,
+    "extends": "airbnb-base",
+    "env": {
+      "node": true,
+      "es6": true,
+      "mocha": true
+    },
+    "rules": {
```
Comment (paradisekelechi) [inline .eslintrc]: @samorahno Please, what informed your choice of these rules?

File: .eslintrc [hunk 8]
```diff
@@ -0,0 +1,33 @@
+{
+    "root": true,
+    "extends": "airbnb-base",
```
Comment (paradisekelechi) [inline .eslintrc]: @samorahno We are using `airbnb` and not `airbnb-base`

**Review by paradisekelechi — CHANGES_REQUESTED** [review]

File: .eslintrc (hunk 7 above)
Comment (samorahno) [reply .eslintrc]: @paradisekelechi, thank you for bringing this to my notice. I found the rules on the pivotal tracker story content, I will go over it again and make use of airbnb

**Review by paradisekelechi — APPROVED** [review]

**Pull request merged**